firmware/src/ledger/signer/test/signer_ux/test_signer_ux.c

```diff
@@ -0,0 +1,113 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2021 RSK Labs Ltd
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+#include <assert.h>
+#include <stdio.h>
+#include "signer_ux.h"
+
+static unsigned int G_mock_idle_time_ms;
+static unsigned int G_mock_screensaver_timeout_ms;
+enum mock_ui_state { MOCK_UI_INFO, MOCK_UI_SCREENSAVER, MOCK_UI_INVALID };
+enum mock_ui_state G_mock_ui_state;
+
+// Helper functions
+static unsigned int max(unsigned int a, unsigned int b) {
+    return a > b ? a : b;
+}
+
+// Mock function calls
+void signer_ux_info(void) {
+    G_mock_ui_state = MOCK_UI_INFO;
+}
+
+void signer_ux_screensaver(void) {
+    G_mock_ui_state = MOCK_UI_SCREENSAVER;
+}
```

Comment:
Sure, yeah. You have a point there. I think we can put this up pretty easily. Will work on it and let you know in case I find it too complicated for some reason.

I agree it's worth adding this since this logic is part of the signer UX, not the SDK. Thanks for the review

Reply:
Pushed a new commit that tackles this suggestion, let me know if it looks good.